var/spack/repos/builtin/packages/dyninst/package.py

```diff
@@ -20,78 +20,98 @@ class Dyninst(Package):
-        if spec.satisfies('@:8.1'):
-            configure("--prefix=" + prefix)
-            make()
-            make("install")
-            return
-
-        libelf = spec['elf'].prefix
-        if spec.satisfies('@:9'):
-            libdwarf = spec['libdwarf'].prefix
-
-        with working_dir('spack-build', create=True):
-            args = ['..',
-                    '-DBoost_INCLUDE_DIR=%s'    % spec['boost'].prefix.include,
-                    '-DBoost_LIBRARY_DIR=%s'    % spec['boost'].prefix.lib,
-                    '-DBoost_NO_SYSTEM_PATHS=TRUE',
-                    '-DLIBELF_INCLUDE_DIR=%s'   % join_path(
-                        libelf.include, 'libelf'),
-                    '-DLIBELF_LIBRARIES=%s'     % join_path(
-                        libelf.lib, "libelf." + dso_suffix)]
-            if spec.satisfies('@:9'):
-                args.append('-DLIBDWARF_INCLUDE_DIR=%s' % libdwarf.include)
-                args.append('-DLIBDWARF_LIBRARIES=%s'   % join_path(
-                    libdwarf.lib, "libdwarf." + dso_suffix))
-            # For @develop + use elfutils libdw, libelf is an abstraction
-            # we are really using elfutils here
-            if spec.satisfies('@develop') or spec.satisfies('@10:'):
-                tbb = spec['tbb'].prefix
-                args.append('-DTBB_INCLUDE_DIRS=%s' % tbb.include)
-                args.append('-DTBB_LIBRARIES=%s'    % join_path(
-                    tbb.lib, "libtbb." + dso_suffix))
-                args.append('-DLIBDWARF_INCLUDE_DIR=%s' % libelf.include)
-                args.append('-DLIBDWARF_LIBRARIES=%s'   % join_path(
-                    libelf.lib, "libdw." + dso_suffix))
-            if spec.satisfies('arch=linux-redhat7-ppc64le'):
-                args.append('-Darch_ppc64_little_endian=1')
-            args += std_cmake_args
-            cmake(*args)
-            make()
-            make("install")
-
-    @when('@:8.1')
-    def install(self, spec, prefix):
-        configure("--prefix=" + prefix)
-        make()
-        make("install")
+    # Versions 9.3.x used cotire, but have no knob to turn it off.
+    # Cotire has no real use for one-time builds and can break
+    # parallel builds with both static and shared libs.
+    @when('@9.3.0:9.3.99')
+    def patch(self):
+        filter_file('USE_COTIRE true', 'USE_COTIRE false',
+                    'cmake/shared.cmake')
+
+    def cmake_args(self):
+        spec = self.spec
+
+        # Elf -- the directory containing libelf.h, either
+        # prefix/include (elfutils), or prefix/include/libelf
+        # (libelf).
+        elf = spec['elf'].prefix
+        if spec.satisfies('@9.3.0:'):
+            elf_include = elf.include
+        else:
+            elf_include = join_path(elf.include, 'libelf')
+
+        # Dwarf -- the directory containing elfutils/libdw.h or
+        # libdwarf.h, and the path to libdw.so or libdwarf.so.
+        if spec.satisfies('@10.0.0:'):
+            dwarf_include = elf.include
```

Comment:
Should this be `dwarf_include = elf_include` or is it always the case that the dwarf headers are in `<elf-base-prefix>/include`?

Reply:
@mwkrentel I wanted to make sure - is this in fact what you intended?

Reply:
Yes, that's what I intended.

elf_include means the location of libelf.h from one of two locations
(include or include/libelf).  elf.include means the top-level include
directory for elfutils.

Dyninst uses `#include <elfutils/libdw.h>`, so it wants the elfutils
top-level include directory, not the elfutils subdir where libdw.h is.
That's why I considered elf.include slightly more accurate.  But
they're the same answer for this case, so it's really six of one.

If you prefer, I could change this to `find_headers('libdw', ...)`.  But
then I'd need `dirname(dirname(find_headers(...)))`  because libdw.h is
in elfutils/libdw.h.  IMO, I felt that wasn't worth the extra code
since you're going to end up with the same answer anyway.

Btw, you could set this to gibberish and it would still work because
the same -I dir it needs is already added from LIBELF_INCLUDE_DIR in
this case.  So, I'd prefer to leave it as is and merge.